### DISCUSSION TITLE
Starting to drop some 32-bit Packages

### DISCUSSION THREAD

**Discussion by lazka** [msys2/MINGW-packages#19326] [General]
This is companion discussion to the recent news entry. If there are any questions let us know.

I've compiled a list of which 32bit packages were downloaded in the last week: https://gist.github.com/lazka/3a894412a32acb583cabbb89fcb11df4

**Comment by mmuetzel** [comment]
Maybe the page hasn't updated for me. But there is no news entry about the imminent removal of 32-bit packages here afaics:
https://www.msys2.org/news/

Edit: It might have been better to announce the removal of these packages at least a couple of weeks *before* starting to remove them. (That's already a very tight time frame given that we are near the end of the year which is often quite busy anyway.)
As it stands, the packages are being removed without any prior notice at all...

**Reply by lazka** [reply]
It's still a draft: https://github.com/msys2/msys2.github.io/pull/299/files I just needed something to link to, so I created this before.

**Reply by mmuetzel** [reply]
Thanks for the pointer.

To clarify: I'm not against dropping support for 32-bit Windows in principle. Just some thoughts:
I don't have any usage stats. But from personal experience, I can tell that especially some university labs are using still quite old hardware. So, it's not entirely clear to me why actively removing 32-bit packages has started with scientific libraries.

Instead of putting effort into actively removing packages for those platforms, would it be acceptable to not put much effort into actively *maintaining* 32-bit packages? E.g., remove a package only if it doesn't rebuild on an update or pkgrel bump and it falls into the criteria you laid out.

**Reply by lazka** [reply]
Yes, I agree that dropping when things fail should be sufficient. I don't have much insight in scientific libraries, but I assumed they weren't used that much outside of MSYS2. I guess I was wrong there (?)

@MehdiChinoune why are you so keen on dropping packages now? Can we maybe pause it and wait for some feedback?

**Reply by lazka** [reply]
Keep in mind that this is pushing open source developers to drop support for a chunk of their user base, which they in many cases have no insight into, and no (easy) way to force them to 64bit. So I agree that it's good to start this, but we should be gentle :)  It's not just porting some code.

**Reply by jeremyd2019** [reply]
iirc Windows 10 (including 32-bit) is still supported by Microsoft until 2025, and I also read recently that Microsoft will be selling ESU support, even to consumers (not just companies as usual) so that could push that timeline even further.

**Reply by mmuetzel** [reply]
> It is part of my plan to pause and wait for feedback.

That pause didn't take very long:
https://github.com/msys2/MINGW-packages/commit/77c55682106b5964b71099051078ac71949dc1c9

**Reply by mmuetzel** [reply]
> Because, It needs qt5-static which I dropped 32bit from it, yesterday.

Given the feedback from here so far, maybe it would be better to revert the change from yesterday instead.

**Reply by mmuetzel** [reply]
Maybe an example helps to make clear why removing all of these packages like this might not be a good idea:
Let's say you have a piece of specialized laboratory equipment for which there are only (proprietary) drivers for 32-bit Windows. Even if that piece of lab equipment was acquired quite some time ago and there won't be newer drivers for it, it might still do its job. That means that you'd be "stuck" with 32-bit Windows if you like to continue using that piece of lab equipment.
Now, let's say you'd like to start a new measurement campaign for which you'd like to implement a couple of new experiments. You might want to use, e.g., the netcdf library for that implementation.
You could have used the package from MSYS2 for this. But now it's going to be removed. It's still possible to build that library locally. But that will get more and more difficult as soon as more and more dependencies disappear.

**Reply by mmuetzel** [reply]
> Q: Which countries do scientific research?
> A: Rich countries.

And other countries are excluded from participating in research???

> => They have enough money to fund their research and upgrade their hardware.

Maybe, in an ideal world. But research funds aren't unlimited. And some lab equipment cannot just be replaced (because they've been produced as part of a past research project).

> You could install 32-bit drivers on 64-bit Windows, Am I wrong?

Can you?

> Get your experimental data from your machines and use 64-bit Software to Analyse and Visualise them.

That's only a work-around solution. What if you'd like to update your *control* software (instead of the post-processing software)?

**Reply by OBattler** [reply]
Just because everyone uses a 64-bit version of Windows, there's not automatically a benefit from compiling 64-bit binaries, unless the program uses a lot of RAM. So for the vast majority of software, the push to 64-bit makes little sense.

**Reply by sskras** [reply]
@MehdiChinoune wrote [1 hour ago](https://github.com/msys2/MINGW-packages/discussions/19326#discussioncomment-7818736):

> Because Those old machines don't produce as accurate results as other foreign labs, They couldn't publish them in a scientific article. 

I think that's misleading. The cause is not age of the machines, but rather functionality. Old machine can calculate 64-, 128- and 256-bit arithmetic/logical operations just fine, only slower.

> Most of people I know in scientific research just use Origin-Lab (Or whatever their institut paid for), they don't use Free/Open Software.

Do you believe that's a justifiable reason to lessen spread of F/OSS in the scientific research area (on older machines)?
I hope you do not :)

> I think the situation you are talking about is unreleastic:
> 
> * An old equipment
> * Requires Windows 10 32-bit to take measurements
> * Use Free/Open software.
> * Use latest updates.

What I see most unrealistic here is Cygwin / MSYS2 dropping support for older Windows versions in the past when there was only little (or even no) additional effort needed to maintain them (the runtime itself) as these OSes stopped being moving targets already (IOW, their API became static forever since the EOL).

@mmuetzel: I guess it's time to revisit @reactOS and @Midipix projects.

Although I don't know how much of an effort would be required to get the 32-bit build of @Midipix.
Quoting the main dev from Libera.Chat#midipix IRC channel:

![image](https://github.com/msys2/MINGW-packages/assets/7887758/4b120510-0eab-44eb-8dec-0dd23904145a)

**Reply by sskras** [reply]
@MehdiChinoune:

> We were talking about experimental machines, not computing machines.

I am interested in an example of the machine age leading to less accurate scientific results.

My understanding was that scientific results come from computing software that does computing operations that runs on specific machine. Still seeing no relation between machine being old and the results being less accurate. Hence my frustration.

**Reply by sskras** [reply]
@MehdiChinoune:

> Their machines produce a 2D images but It doesn't compete with latest machines that produce 3D images.
  [...]
> The latest machines are more technologically advanced, you may complete an experiment within an hour that could take 2 days with older machines.

Thanks for the example (I may ask you for their direct contact later).

But these are two different statements. One mentions accuracy, another one mentions the speed. Do you mean that lesser accuracy is a consequence of an operator (eg. a research lab) adding timing restrictions? In that case the arguments are not about fundamental difference in accuracy, just about a fundamental difference in timing. And as this seems not to be a realtime calculations, I think we proved that your original statement wasn't very exact.

IOW, the thing is more about decision to _restrict the processing time_. Which is very different from the question whether **32-bit software can calculate results that would be as accurate** as results from 64-bit software.

**Reply by OBattler** [reply]
> Users won't stop downloading and using 32bit unless you push them to do.

And why would you push the users to do that? What benefit do they get from that? Not that regular users care about msys2 packages - Windows isn't Linux, most users just download the binary of a program and run it. Which is another reason this is going to fail if the intention truly is to push users to only use 64-bit applications, also because nothing stops a user from sticking to an older version of an application.

**Comment by jeremyd2019** [comment]
I was just wondering, is there a way to find out from `pacman` when installed packages are removed from the repos?  Ie, is there some way to give an automated notification when users update that *hey, be warned, something you have installed won't be getting any more updates*.

Based on my past experience with the deprecation of 32-bit msys2, I only noticed that I wasn't getting any more msys updates when running pacman -Syu, and eventually went looking for the reason why and found the announcement of the deprecation.  If it was only *some* packages, I might not have noticed, or at least it would have taken a lot longer.

**Reply by Biswa96** [reply]
> is there a way to find out from pacman when installed packages are removed from the repos?

`pacman -Qm`

**Reply by lazka** [reply]
We could add a dummy package to "base", which shows a `pacman -Qm` report in the post-install script, and which we bump from time to time.

The above command is also part of the "new" docs page on updating: https://www.msys2.org/docs/updating/

**Reply by lazka** [reply]
I've added the command to the news entry draft

**Comment by MehdiChinoune** [comment]


**Reply by mmuetzel** [reply]
How is that better than the earlier proposal above?
Copying here for convenience:
> Instead of putting effort into actively removing packages for those platforms, would it be acceptable to not put much effort into actively maintaining 32-bit packages? E.g., remove a package only if it doesn't rebuild on an update or pkgrel bump and it falls into the criteria you laid out.

> Maintainers look at their situations and decide if It's worthy or not.

Maybe, it's just me. But I don't like that "totalitarian" approach. Users would need to come to the round of "almighty maintainers" which will decide if their quest is "worthy" or not...

**Reply by lazka** [reply]
there is no such law. and building one time centrally is less resource intensive then forcing other people to build multiple times, especially on shared server hardware like GHA vs desktop machines.

**Reply by lazka** [reply]
I think having a generic template for building for a new environment might be a good idea. It doesn't have to be 32-bit specific.

**Reply by davidgiven** [reply]
> We agreed on the above proposal years ago, but people reject/refuse to drop 32-bit and insist on fixing issues with it.

Surely this means that people want to keep the 32 bit binaries?

**Reply by jeremyd2019** [reply]
> > We agreed on the above proposal years ago, but people reject/refuse to drop 32-bit and insist on fixing issues with it.
> 
> Surely this means that people want to keep the 32 bit binaries?

Not only that, but it also means that there are people willing to volunteer time/effort to keep them working.

**Reply by mmuetzel** [reply]
> > Not only that, but it also means that there are people willing to volunteer time/effort to keep them working.
> > 
> 
> So, let them show up. We only see them complaining.

You are not really inviting anyone to do that though. Instead you are starting a fight with anyone that does up texting to fix anything 32-bit...

You really need to make up your mind about what you want: Welcome people white would like to help with the support for that target? Or fight them off?

**Comment by OBattler** [comment]
You have just inconvenienced the 86Box project by removing 32-bit qt5-static... our main Windows binaries are 32-bit because they perform better. We have no problem finding ways to improve the performance of 64-bit binaries but we need time, though that assuming it can be achieved and it's not simply a result of the architecture itself (longer pointers, fewer direct addressing instructions, etc.). Right now, you're basically pulling the rug from under our feet.

**Reply by OBattler** [reply]
No formal benchmarks other that the emulator's internal emulated CPU frame counter, which shows there being a few % less performance on the 64-bit binary. But I'll see what I can do as far as benchmarks are concerned.

**Reply by OBattler** [reply]
On macOS, we have no choice - Apple completely dropped support for 32-bit binaries long ago.

And similarly on Linux - I've yet to see a single of our Linux users use a 32-bit version of Linux, and running a 32-bit binary on 64-bit Linux is not as easy as running a 32-bit binary on 64-bit Windows. And it's also a different case - the difficulty has been there since the beginning, they never really allowed easy mixing of binary bitness, it's a bit different on Windows, where such mixing is openly supported by the OS itself, and there's in fact plenty of software that is in no rush to provide a 64-bit version.

And yes, I'm talking about 86Box.

**Reply by mmuetzel** [reply]
For Linux distributions, there are many alternatives to chose from. Some of them still package for 32-bit platforms (e.g., Debian, Alpine, ...).
For Windows, there aren't as many binary distributors to chose from...

**Reply by OBattler** [reply]
@MehdiChinoune So if Linux developers threw themselves from a window, you would as well? Please, for the love of God, get a better argument instead of "but Linux also did this". Also, as mmuetzel showed, there are still distributions that package 32-bit, on Windows, we'd all be relegated to Microsoft Visual Studio and its vcpkg. So you'd effectively be ensuring the strengthening of Microsoft's position.

**Reply by OBattler** [reply]
Given that most of my users download 86Box directly from our Jenkins (https://ci.86box.net), that's not accounted for by the GitHub statistics. And there, plenty more on Windows download the 32-bit version, especially since we officially recommend it, for reasons I mentioned above. Even I personally have been using the 32-bit version on Windows, as do the rest of my team. We just never download it from GitHub.

**Reply by OBattler** [reply]
A lot of us just go to Jenkins by default, and we often also outright tell people to try the latest nightly build when they report issues.

> Also, I don't see any recommendation on the website about x86 (32-bit) binaries!

That will be rectified. Though, we do also have a Discord server where we clearly tell people 32-bit binaries are preferred.

**Reply by mmuetzel** [reply]
We could have taken a page out of their book when it comes to deprecating platform support. They are announcing two years in advance to drop building the kernel.
However, they currently plan to still build and ship all packages for i686 unless maintainers would like to drop specific packages...

**Reply by mmuetzel** [reply]
The Octave community doesn't distribute binaries of Octave in general. It only develops the source code. Binaries are made by distributors.
The sole exception is Windows.

I was hoping that Octave could move away from that exception by pointing Windows user to MSYS2. The 32-bit binary were meant to be serving as a test-bed for that. (There are probably less users. So, the impact might have been lower.)
With the recent developments, I'm no longer sure whether this is a feasible plan though...

**Reply by sskras** [reply]
Now that was hostile.

**Reply by mmuetzel** [reply]
> So, I could go ahead and drop 32-bit from octave!

Please don't.

**Comment by lazka** [comment]
Some changes I've made:

* Added a issue template for requesting an environment for a package: https://github.com/msys2/MINGW-packages/issues/new?assignees=&labels=package-request&projects=&template=environment_request.yml&title=Build+[package+name]+for+[environment]
* Made the NEWS draft point to the issue template instead of here, so things are more focused
* I've restored various packages in the removal list which had more than 2 downloads in the last 9 days

What I'm planning to do:

* Include some download stats on packages.msys2.org so everyone can easily see if there are any users for a package, in case issues arise with that package. The download stats are already provided by mirrorbits, but we can make them more easily accessible/visible (https://mirror.msys2.org/msys/x86_64/gnupg-2.2.41-1-x86_64.pkg.tar.zst?stats). These are not that exact as the ones I posted above, since they also consider non-pacman downloads, but better than nothing.

**Reply by lazka** [reply]
hm, true, that's a bit more tricky.

**Comment by davidgiven** [comment]
I have several projects designed to run either with or on old machines, and people actually use them. After running into mysterious problems with packages disappearing while working on one, I've tried the most common one people actually use, and it fails too. Please reconsider? This is going to cause me significant work. Dropping 32 bit would kinda defeat one of the major points of this particular project.

**Reply by lazka** [reply]
Could you give us a list of packages that you are missing now?

**Reply by davidgiven** [reply]
The packages I've found so far are:

qt5-base
qt5-static
glfw
abseil-cpp

In terms of other large packages that might disappear in the future, I also depend on wxwidgets, libusb, sqlite3. Plus of course a handful of smaller packages. 

Technically there's also protobuf, but currently I have to install an old version of this which doesn't use abseil because of an outstanding bug where the abseil-cpp package doesn't contain static libraries (#17852). Currently I build a single portable executable for ease of use. I was actually thinking of abandoning the single executable and using dynamic linking due to another outstanding bug where qt5-static has a bad pkg-config file (#19690), but then the abseil-cpp package vanished while I was working on it...

**Reply by davidgiven** [reply]
> abseil-cpp is not an old project to expect someone to rely on!

I don't follow you here?

**Reply by sskras** [reply]
@MehdiChinoune writes:

> Projects that typically designed for old OSs rely on old projects, because new projects default to use Latest APIs.

Nothing prevents new project to have branching between old and new APIs, both having in place.
Nothing is able to stop that except power of the human brain. ;)

**Reply by sskras** [reply]
@davidgiven: Looks like we, the minority are forced to be expats here.
Luckily the [Midipix](https://news.ycombinator.com/item?id=9651011) project doesn't reject the idea of running on 32-bit machines. 

It's just not ready yet (due to the manpower shortage, I guess).

**Reply by davidgiven** [reply]
> Projects that typically designed for old OSs rely on old projects, because new projects default to use Latest APIs.

I don't understand your reasoning here. If the project works, it works. If it doesn't, it doesn't. Rejecting a project because you think it might not work in the future isn't achieving anything and smells heavily of gatekeeping.

Re building 64-bit on Linux: the situation there is not remotely comparable. I do not distribute Linux binaries (you didn't notice that the Autorelease workflow only targets Windows and OSX). Linux users either build from source or rely on a third party distribution to do so. WordGrinder is in most distributions and is built for a huge number of different architectuyres.

Re building 64-bit on OSX: I would if I could. People ask me for it. Unfortunately I cannot find any CI build system which will do so as Apple are trying very very hard to deprecate i386 machines and all available VMs I can find don't have i386 tooling. (The same also applies to PowerPC machines. Yes, people have asked about this too.)

...

Regardless of my application, there are lots of 32-bit machines around, and there are people who still want to get useful work done on them. Some of them are in actual industrial use. People need a toolchain to do support these. As far as I know, mingw is the _only_ such toolchain with decent package support which allows building 32-bit windows binaries. If mingw drops 32-bit, then realistic development can't be done on these machines any more. Is your plan to just abandon them? Because I think that's really dangerous.

**Reply by davidgiven** [reply]
Not that weird --- there's a new LLVM C compiler for the 6502, and you should see some of the platforms Rust supports...

Re Windows versions, I can't give you figures --- I don't collect statistics. The main application I'm concerned about is WordGrinder, which right now runs all the way back to XP. I do receive emails from users telling me how nice it is they can use it on old machines. Most of these machines are more modern versions of Windows, of course, but a lot of them are on 32-bit hardware. A lot of these emails come from older people. FluxEngine targets more recent machines, but it's still really convenient to run on older hardware because they have more ports than newer ones, and those of course tend to be 32-bit systems.

But I don't see why that's relevant. Why did you ask?

> Are you expecting the community to support what the industry dropped.

No, you've missed my point. What I think is dangerous to promote waste, needless obsolescence, and the destruction of knowledge and history. If there was any other viable win32 toolchain, that'd be one thing, but (as far as I know --- _please_ correct me if I'm wrong) there isn't. Once the ecosystem and tooling is destroyed, it'll never come back.

**Reply by jeremyd2019** [reply]
As far as I know, there are no plans from mingw-w64 to drop support for 32-bit targets.  The "toolchain" is still intact.

What I see here is basically arguing past each other, one side citing "waste" of cloud cycles, storage, and bandwidth producing 32-bit packages that they see as unnecessary/obsolete, and the other citing "waste" of old but still-working systems.  Neither side seems convinced by the other.

For me, I don't see the "wasted" cloud resources as that big of an issue, and I do still run a 32-bit Windows 10 tablet that is definitely showing its age.  I also take the time and effort myself to keep building updated 32-bit msys2 packages even after this project dropped support for that (https://github.com/jeremyd2019/msys2-build32/releases/repo) - so far only msys2-runtime cannot be updated (because Cygwin upstream actively ripped out all the 32-bit support code in version 3.4 when they dropped support).  In the end, if someone (or enough someones) care enough, and if the staunchly anti-32-bit members of this project cannot be convinced or overruled, something similar could happen with 32-bit MINGW packages: you can build and release packages (even solely on free github resources), and provide an additional pacman repository that people could drop into pacman.conf to get them.  I hope it won't come to that though, and it sounds like there is support for restoring 32-bit packages on a case-by-case basis if downstream projects need them.  Not everyone will agree on the definition of "need" however.

**Comment by MehdiChinoune** [comment]


**Reply by davidgiven** [reply]
Sorry, I misread your question. I thought you wanted to know about the old systems, not the news ones. In terms of new systems, most of my 32 bit users use Windows 10 (naturally).

> 32-bit systems run in general on Windows<8.1

You knows Windows 10 32-bit is still in active service and supported by Microsoft, right? It goes EoL in 2025 (not counting any potential LTS for enterprise users).

> I have restored mingw32 for abseil-cpp and protobuf.

Thank you!

**Comment by brunvonlope** [comment]
@MehdiChinoune or @lazka According to Wikipedia, 32bit Windows 10 will survive like a zombie up to **2033**. So, it's not too accurate in the news page saying that 32bit will somewhat die in 2-3 years. I suggest fixing it.

All that being said, I dont understand the surprise of so many people here about 32bit removal. Only Windows maintains this "thing" in the mainstream OSes and as a Windows user I seriously doubt how well a machine with 4GB runs Win 10 22H2.

**Reply by lazka** [reply]
> @MehdiChinoune or @lazka According to Wikipedia, 32bit Windows 10 will survive like a zombie up to **2033**. So, it's not too accurate in the news page saying that 32bit will somewhat die in 2-3 years. I suggest fixing it.

I've clarified it: https://github.com/msys2/msys2.github.io/commit/306e137f051307fdbeb9def2673e1f8439649855

**Comment by jonoomph** [comment]
Sorry, a little late to the party here. I recently asked a question https://github.com/msys2/MINGW-packages/issues/20090 about missing 32-bit packages, which OpenShot Video Editor uses in our build system. I wanted to share my feedback here as well. 

While I understand that MSYS2 might show near-zero download stats for some of these 32-bit packages, like Qt5, PyQt5, etc... 32-bit binaries produced by OpenShot's build server are still being downloaded hundreds of thousands of times, primarily in the education sector (which is famous of running ancient versions of Windows and old hardware). We don't upgrade these packages very often (due to fears like this, where packages will be gone), but there is still a very large need in the world for good 32-bit packages. Thanks!

**Reply by jonoomph** [reply]
Actually, I claim ignorance for the exact setup of these 32-bit systems. All I know is people keep downloading 32-bit OpenShot, we had more than 10k downloads from Windows 7 machines in the past 30 days, and I continue to get emails from school districts wanting to run OpenShot on old computers.

**Comment by pm215** [comment]
Hi -- I see that the mingw-w64-i686-dtc package has just been removed. This broke QEMU's CI, which checks we can build on the 32-bit mingw platform. (This isn't the first of the packages we use that got removed -- libusb and spice also vanished recently -- but those are optional extras so we tweaked the CI to stop trying to install them. But dtc is more fundamental.)

Should we see this as "MSYS2 is not actually supporting the i686 platform any more, stop using it" ? The NEWS item suggests you're only dropping "Packages which are likely not used outside of MSYS2", but I would suggest that libraries are among the things that third-parties using MSYS2 as a build target are likely to be using.

**Reply by pm215** [reply]
32-bit Windows is definitely pretty much on the edge of our supported platform set, certainly, though our deprecate-and-drop cycle is clearly a bit longer than yours is :-) In the interim we can build test at least on the Fedora mingw32 setup, though we would lose the runtime testing part of our CI.

(I dunno about QEMU's upstream users, but MSYS2's 32-bit QEMU package clearly did have users, because one of them filed https://github.com/msys2/MINGW-packages/issues/19200 when it broke.)

**Comment by MehdiChinoune** [comment]


**Reply by davidgiven** [reply]
> many of them want to keep supporting Old OS/Hardware (Windows 7 32-bit) which was dropped a year ago

No, Window 10 32-bit is still in active support from Microsoft and will be until 2025.

**Reply by sskras** [reply]
@MehdiChinoune wrote 46 minutes ago:

> When Linux distributions dropped 32-bit they keep only some few packages

You put it like these were all Linux distros.
No, some Linux distros are keen on supporting 32-bit x86 platforms. Eg. @AdelieLinux does i586 (pmmx) builds.

**Reply by sskras** [reply]
> major Linux distributions

Majority of them has become very much Windows-alike where the infamous Embrace-Extend-Extinguish (EEE) strategy is still much supported.

A [questionable thing](https://pygear.com/eat-shit-a-billion-flies-cant-be-wrong/) to use as an argument. But yes, YMMV.

**Reply by davidgiven** [reply]
> > No, Window 10 32-bit is still in active support from Microsoft and will be until 2025.
> 
> I know, I said many not all. You hardly find someone using Windows 10 32-bit.

That is true for consumer users, but not at all for industrial users. And you keep referring to the now obsolete Windows 7 as the example of the OS for 32-bit hardware, but it's important to remember that Microsoft still supports it with current operating systems.

Also --- you say that demand for 32-bit libraries comes from developers; does Mingw even _have_ non-developer users?

**Comment by jonoomph** [comment]
For those interested in **exact** download stats *(Apr 2023 to Feb 2024)*, here are the download #'s for the OpenShot Video Editor 32-bit installer. Notice that the user must really hunt for a 32-bit download on our website, as the default is 64-bit.

![OpenShot-Windows-Download](https://github.com/msys2/MINGW-packages/assets/5551883/42ce5507-e15f-47cc-a6bc-d71cf5bad6f8)

**32-bit download stats** (I was quite shocked in the # of downloads for 32-bit):
![OpenShot-3-1-1-Download-Stats-32-bit](https://github.com/msys2/MINGW-packages/assets/5551883/9dfc354a-41d5-4e70-ba16-719d1886e330)

**Reply by lazka** [reply]
So that makes ~8%.

I'm wondering how large the overlap is between 32 bit and Windows 7 users since things like Python no longer run there for example (upstream dropped support after Python 3.8, we kept it somewhat working until Python 3.10 via extra patches and reverts)

**Comment by Jehan** [comment]
Hi all!

`lua-lgi` was removed from MSYS2 (as 32-bit) because [considered a leaf package](https://github.com/msys2/MINGW-packages/commit/93f22f99e2471f54103ba6e285e2f1c86f7766f2#diff-6a773edb90ff26b85cc0c6742490a51aa43712a1077297f7cd622c73f8cb3060).

Your news about removing 32-bit packages was saying:

> Leaf packages with complex and resource intensive builds that are likely not used

But `lua-lgi` is being used by upcoming GIMP 3. Lua is one of the officially supported script language, through GObject Introspection binding (i.e. with `lua-lgi`).

Would it be possible to bring this package back please?
Thanks!

**Reply by lazka** [reply]
https://github.com/msys2/MINGW-packages/commit/afd4594ab3762c1c189a350fa6d3b06675f68678

**Reply by Wormnest** [reply]
Note that this was one of the reasons that I asked a few weeks ago on Discord to please reconsider dropping the 32-bit gimp3 build as a way of noticing build failures.

**Reply by Wormnest** [reply]
Ah, thank you lazka 😃

**Reply by Jehan** [reply]
Awesome. Thanks @lazka !

**Reply by lazka** [reply]
I'd suggest you think about 32bit going away eventually though.

**Reply by Jehan** [reply]
Sure. We are discussing this on IRC just now. (you aren't on our IRC channel these days! 😢)

Right now, the direction we seem to come into an agreement is that we are thinking maybe making a few GIMP 3 releases with 32-bit, so that 32-bit people at least have the option to use GIMP 3. And also to give them time to prepare by making an announcement (like we'll drop 32-bit after X micro-version releases).

@Wormnest is also planning on working on a WIA plug-in to replace TWAIN soon after we release GIMP 3 (since the 32-bit TWAIN plug-in is used even in 64-bit GIMP and is one of the last big reason why we need 32-bit binaries even for our 64-bit build). Then if it pans out as we hope, we won't have a time range with no scanner plug-in (feature regression).

**Reply by lazka** [reply]
Joining the GIMP IRC server results in an TLS error (`Connection closed unexpectedly: Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: Host: [irc.gimp.org](http://irc.gimp.org/). is not in the cert's altnames: DNS:irc.acc.umu.se, DNS:irc-services.acc.umu.se, DNS:irc1.acc.umu.se, DNS:irc2.acc.umu.se, DNS:irc3.acc.umu.se, DNS:irc4.acc.umu.se`), and trying to join the marix channel results in `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://matrix-client.matrix.org/_matrix/client/v3/directory/room/%23gimp%3Agnome.org. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 502.`. So it's not that easy :)

**Reply by Jehan** [reply]
Ah yeah there are TLS issues lately, and I think admins on GIMPnet are procrastinating a bit on fixing it. We set `ssl_verify = "no";`  (in irssi, but other clients have same settings somewhere usually) these days.

As for the Matrix server, I can't say but anyway it's not synced with IRC anymore anyway (for a few months now, GNOME removed the IRC-Matrix bridge) and not sure if it will come back. Some devs are also on Matrix. Me I'm subscribed, but don't read. I only join when I get a notification in my inbox.

Yeah I know, it's messy. 🤨

**Comment by jeremyd2019** [comment]
> I do still run a 32-bit Windows 10 tablet that is definitely showing its age.

Well, this tablet had some sort of hardware problem, it will no longer boot *anything* (its OS, usb stick, ... just goes to efi setup).  So I'm less concerned about 32-bit packages than I was a week or two ago 😉

I was able to recover the 500gb hard drive in the dock, but I have not figured out yet how to get at the data on the onboard 64gb eMMC.

**Reply by revelator** [reply]
damn newer develop on something with hard soldered components 🤕 to many headaches when something goes wrong.

**Comment by selectstriker2** [comment]
Is the general intention is to drop support for the MINGW32 build system / environment entirely?

**Reply by revelator** [reply]
mostly i guess, seems everyone think 32 bit is dead or atleast hope it will soon be xD.
